barman/server.py

```diff
@@ -1705,6 +1705,16 @@ def backup(self, wait=False, wait_timeout=None, backup_name=None, **kwargs):
             if not previous_backup:
                 self.backup_manager.remove_wal_before_backup(backup_info)
 
+            # check if the backup chain (in case it is a Postgres incremental) is consistent
+            # with their checksums configurations
+            if not backup_info.is_checksum_consistent():
+                output.warning(
+                    "This is an incremental backup taken with `data_checksums = on` whereas "
+                    "some previous backups in the chain were taken with `data_checksums = off`. "
+                    "This can lead to potential recovery issues. Consider taking a new full backup "
+                    "to avoid having inconsistent backup chains."
+                )
+
```

Comment:
I thought of putting the backup check on the `PostgresBackupExecutor`, since it is specifically for Postgres incremental backups, but I think the warning has more chances of not being ignored if displayed at the end, so I put it on the server backup method instead.

Reply:
I think that makes sense -- that way it's less prone to being missed/ignored.